usecases/memwatch/monitor_test.go

```diff
@@ -159,37 +159,43 @@ func TestMappings(t *testing.T) {
 		mappingsLeft := getMaxMemoryMappings() - currentMappings
 		assert.InDelta(t, mappingsLeft, addMappings, 10) // other things can happen at the same time
 		path := t.TempDir()
+
+		limitReached := false
+
 		// use up available mappings
-		for i := 0; i < int(mappingsLeft)+5; i++ {
+		for i := 0; i < int(mappingsLeft)*2; i++ {
 			m.Refresh(true)
 			file, err := os.OpenFile(path+"example"+strconv.FormatInt(int64(i), 10)+".txt", os.O_CREATE|os.O_RDWR, 0o666)
 			require.Nil(t, err)
-			defer file.Close()
+			defer file.Close() // files should stay open until end of test to continue to use mappings
 			_, err = file.Write([]byte("Hello"))
 			require.Nil(t, err)
 
 			fileInfo, err := file.Stat()
 			require.Nil(t, err)
 
-			// there might be other processes that use mappings
+			// there might be other processes that use mappings. Don't check any specific number just that we have
+			// reached the limit
 			if mappingsLeft := getMaxMemoryMappings() - getCurrentMappings(); mappingsLeft <= 0 {
-				require.NotNil(t, m.CheckMappingAndReserve(1, 0))
+				limitReached = true
+				break
 			} else {
-				require.Nil(t, m.CheckMappingAndReserve(1, 0))
 				data, err := syscall.Mmap(int(file.Fd()), 0, int(fileInfo.Size()), syscall.PROT_READ, syscall.MAP_SHARED)
 				require.Nil(t, err)
 
 				defer syscall.Munmap(data)
 			}
 		}
+		require.True(t, limitReached)
```

Comment:
**question (if-minor):**  won't that fail in case of `else` path?

Reply:
if we only go into the else clause yes.

The idea is:
- see how many mappings we have available
- open many files (and keep them open) to use up available mappings and hit the limit
- assert that we hit the limit

Reply:
gotcha, one suggestion. if possible. could you add a comment explain that this shall be satisfied